MoreLinq/HasDuplicates.cs

```diff
@@ -0,0 +1,27 @@
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        ///   Checks if an enumerable of T contains duplicates
+        /// </summary>
+        /// <param name="source">The enumerable of T to validate.</param>
+        /// <param name="comparer">The comparer used for equality. If none is provided, <see cref="EqualityComparer{T}.Default"/> is used</param>
+        /// <typeparam name="T">The item type</typeparam>
+        /// <returns>true if any element of the source enumerable is duplicated, false otherwise</returns>
+        public static bool HasDuplicates<T>(this IEnumerable<T> source, IEqualityComparer<T>? comparer = null)
```

Comment:
This should follow the pattern of other operators in both System.Linq and MoreLinq where there are two overloads, one with `IEqualityComparer<T>? comparer` and one without. The one without can call the one with using `null` as a parameter.